python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -437,12 +480,32 @@ def current_size(self) -> int:
         return self.num_pending_actors() + self.num_running_actors()
 
     def num_running_actors(self) -> int:
-        return len(self._num_tasks_in_flight)
+        return len(self._running_actors)
+
+    def num_restarting_actors(self) -> int:
+        return sum(
+            running_actor_state.is_restarting
+            for running_actor_state in self._running_actors.values()
+        )
 
     def num_active_actors(self) -> int:
         return sum(
-            1 if num_tasks_in_flight > 0 else 0
-            for num_tasks_in_flight in self._num_tasks_in_flight.values()
+            1 if running_actor_state.num_tasks_in_flight > 0 else 0
+            for running_actor_state in self._running_actors.values()
+        )
+
+    def num_alive_actors(self) -> int:
+        """Alive actors must have inflight tasks and each of those should be in ALIVE
+        state.
```

Comment:
I think "num_tasks_in_flight = 0" should also count towards alive?

Reply:
+1 

Otherwise this is no different from `num_active_actors`